### PR TITLE
HADOOP-17799. Improve the GitHub pull request template

### PR DESCRIPTION
**Pull request by aajisaka**
### Description of PR

Improve the GitHub PR template for developers.

### How was this patch tested?

N/A

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 43s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  32m  8s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  26m  6s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  72m 21s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  21m  5s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  21m 51s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  36m 17s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 39s |  |  The patch does not generate ASF License warnings.  |
|  |   | 132m 48s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3277/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3277 |
| Optional Tests | dupname asflicense mvnsite codespell markdownlint |
| uname | Linux dc7eea988a1c 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / b83e609a92fc34e0f1ab0aab5f54ef1aeb1ba053 |
| Max. process+thread count | 724 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3277/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  32m  7s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  26m  8s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  71m 38s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  20m 51s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  21m 51s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  36m 10s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 45s |  |  The patch does not generate ASF License warnings.  |
|  |   | 131m 32s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3277/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3277 |
| Optional Tests | dupname asflicense mvnsite codespell markdownlint |
| uname | Linux aa347f7e8fe0 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / b83e609a92fc34e0f1ab0aab5f54ef1aeb1ba053 |
| Max. process+thread count | 689 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3277/2/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+
+### Description of PR
+
+
+### How was this patch tested?
+
+
+### For code changes:
+
+- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
```
Comment (virajjasani) [inline .github/pull_request_template.md:16]: Shall we also add this point `PR title starts with JIRA issue id, e.g., 'HADOOP-17799. Your PR title ...'.`?

Comment (jojochuang) [reply .github/pull_request_template.md:16]: probably restore the lines that were removed in this PR.

Comment (aajisaka) [reply .github/pull_request_template.md:16]: Thank you. Added a checklist to make sure the PR title is in the specified format: 458ae01

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 59s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  33m  6s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  24m 57s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  73m 55s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m 36s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  20m 51s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  36m 46s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 37s |  |  The patch does not generate ASF License warnings.  |
|  |   | 136m 37s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3277/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3277 |
| Optional Tests | dupname asflicense mvnsite codespell markdownlint |
| uname | Linux 8802da179697 4.15.0-143-generic #147-Ubuntu SMP Wed Apr 14 16:10:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 458ae01ae5a537bcfccd4bcd5a76ebb7040c2ec2 |
| Max. process+thread count | 561 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3277/3/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Review by virajjasani — APPROVED** [review]
+1 (non-binding). Thanks @aajisaka

**Review by cxorm — APPROVED** [review]
Thanks @aajisaka for improving it, +1
New template is developer friendly.

**Comment by aajisaka** [comment]
Merged. Thank you @virajjasani @cxorm @jojochuang for your review.

**Pull request merged**